olclient/bots.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+
+"""
+Classes and methods for bulk edits that match the following pattern:
+1. Search data dump for X condition
+2. Remove or modify record that fits condition X
+3. Create log of modification
+4. Repeat 1-3 for all records that meet condition X
+"""
+
+
+import argparse
+import datetime
+import logging
+import sys
+
+from olclient.openlibrary import OpenLibrary
+from os import makedirs
+
+
+class BaseBot(object):
+    def __init__(self, ol=None, dry_run=True, limit=1) -> None:
+        """Create logger and class variables"""
+        if ol is None:
+            self.ol = OpenLibrary()
+        else:
+            self.ol = ol
+
+        self.parser = argparse.ArgumentParser(description=__doc__)
+        self.parser.add_argument('-f', '--file', type=str, default=None, help='Path to file containing input data')
+        self.parser.add_argument('-l, --limit', type=int, default=1,
+                                 help='Limit number of edits performed on external data.'
+                                      'Set to zero to allow unlimited edits.')
+        self.parser.add_argument('-d', '--dry_run', type=self.str2bool, default=True,
+                                 help='Execute the script without performing edits on external data.')
+        self.args = self.parser.parse_args()
+        self.dry_run = getattr(self.args, 'file', None) or dry_run
+        self.limit = getattr(self.args, 'limit', None) or limit
+        self.changed = 0
+
+        job_name = sys.argv[0].replace('.py', '')
```

Comment:
```suggestion
        job_name = os.path.splitext(sys.argv[0])[0]
```
https://docs.python.org/3/library/os.path.html#os.path.splitext

Reply:
No longer applicable